### PR TITLE
UI READ support for models and versions

### PR DESCRIPTION
**Pull request by jamieknight-db**
**PR Checklist**

- [x ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Users can browse models in the UI, view details of models, including versions. Version details page displays version data (created_by and updated_by coming soon with access control changes).

Screenshot 2024-09-10 at 12 15 43 PM

Screenshot 2024-09-10 at 12 16 02 PM

Screenshot 2024-09-10 at 12 16 15 PM

Screenshot 2024-09-10 at 12 36 31 PM

### CODE REVIEW

**Comment by jamieknight-db** [comment]
@yc-shawn I am still unable to add you as a reviewer (cc @dennyglee) but would appreciate your review on this one as well!

File: ui/src/pages/VersionDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+} from '../hooks/models';
+import ModelSidebar from '../components/models/ModelSidebar';
+import { formatTimestamp } from '../utils/formatTimestamp';
+
+interface DataType {
+  versionKey: string;
+  versionValue: string;
+}
+
+export default function VersionDetails() {
```
Comment (yc-shawn) [inline ui/src/pages/VersionDetails.tsx:20]: nit: Version is a generic name, can we change this to ModelVersionDetails

File: ui/src/hooks/models.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  });
+}
+
+interface ListModelVersionsParams {
+  catalog: string;
+  schema: string;
+  model: string;
+}
+
+export function useGetModelVersions({
```
Comment (yc-shawn) [inline ui/src/hooks/models.tsx:99]: useListModelVersions

File: ui/src/hooks/models.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+interface GetVersionParams {
+  catalog: string;
+  schema: string;
+  model: string;
+  version: string;
+}
+
+export function useGetVersion({
```
Comment (yc-shawn) [inline ui/src/hooks/models.tsx:123]: useGetModelVersion

File: ui/src/pages/ModelDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  if (!schema) throw new Error('Schema name is required');
+  if (!model) throw new Error('Model name is required');
+  const navigate = useNavigate();
+  const { data } = useGetModel({ catalog, schema, model });
+  const { data: versionData, isLoading } = useGetModelVersions({
+    catalog,
+    schema,
+    model,
+  });
+  console.log('versions', versionData);
```
Comment (yc-shawn) [inline ui/src/pages/ModelDetails.tsx:33]:
```suggestion
```

File: ui/src/pages/ModelDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            }
+            columns={[
+              {
+                title: 'Status',
+                dataIndex: 'status',
+                key: 'status',
+                width: '10%',
+                align: 'center',
+                render: (status) => {
+                  switch (status) {
```
Comment (yc-shawn) [inline ui/src/pages/ModelDetails.tsx:82]: worth having a separate component, might be able to re-used in modelVersionDetails

**Review by yc-shawn — COMMENTED** [review]
Overall looks good

**Review by kriscon-db — APPROVED** [review]
LGTM from my amateur UI code review eyes.

**Pull request merged**